### PR TITLE
Refactor/rewrite version api typescript

### PR DESCRIPTION
**Pull request by kajm**
## Explain the changes you’ve made
Rewrite version-api service to Typescript

## Explain why these changes are made
Having our code written in Typescript makes the development experience better and also improves testability.

## How to test

Concrete example:

1. Checkout this branch
2. run `sls deploy` within service folder for `service version-api`
3.

### CODE REVIEW

File: services/version-api/src/lambdas/getStatus.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  return SUPPORTED_OS.includes(deviceOS);
+}
+
+export async function getStatus(input: LambdaRequest, dependencies: Dependencies) {
+  const userAgent = input.headers[USER_AGENT];
+  const [, deviceApplicationVersion, deviceOS] = userAgent.split(SPLIT_SYMBOL);
+
+  const isMalformedUserAgent = !isSupportedOs(deviceOS) || !deviceApplicationVersion;
+  if (isMalformedUserAgent) {
+    log.error(
```
Comment (dannil76) [inline services/version-api/src/lambdas/getStatus.ts:46]: Change to the updated log.writeError(...)

**Review by dannil76 — CHANGES_REQUESTED** [review]

File: services/version-api/src/lambdas/getStatus.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import * as response from '../libs/response';
+import config from '../libs/config';
+import params from '../libs/params';
+import log from '../libs/logs';
+
+import { getApplicationUpdateStatus } from '../helpers/getApplicationUpdateStatus';
+
+const USER_AGENT = 'User-Agent';
+const SPLIT_SYMBOL = '/';
+const SUPPORTED_OS = ['ios', 'android'];
```
Comment (dannil76) [inline services/version-api/src/lambdas/getStatus.ts:10]: Put consts after types and interface declarations?

**Review by dannil76 — CHANGES_REQUESTED** [review]

File: services/version-api/src/lambdas/getStatus.ts (hunk 2 above)
Comment (kajm) [reply services/version-api/src/lambdas/getStatus.ts:10]: Yes, constants should be after interfaces and types

**Review by dannil76 — APPROVED** [review]

File: services/version-api/src/lambdas/getStatus.ts (hunk 1 above)
Comment (kajm) [reply services/version-api/src/lambdas/getStatus.ts]: fixed

**Pull request merged**